src/dom/Decimal.ts

```diff
@@ -66,26 +65,27 @@ export class Decimal extends Value(
       ignoreAnnotations?: boolean;
       ignoreTimestampPrecision?: boolean;
       onlyCompareIon?: boolean;
+      coerceNumericType: boolean;
     } = {
       epsilon: null,
       ignoreAnnotations: false,
       ignoreTimestampPrecision: false,
       onlyCompareIon: true,
+      coerceNumericType: false,
     }
   ): boolean {
     let isSupportedType: boolean = false;
     let valueToCompare: any = null;
-    if (options.onlyCompareIon) {
-      // `compareOnlyIon` requires that the provided value be an ion.dom.Decimal instance.
-      if (other instanceof Decimal) {
-        isSupportedType = true;
-        valueToCompare = other.decimalValue();
-      }
-      if (other instanceof Float) {
-        isSupportedType = true;
-        valueToCompare = new ion.Decimal(other.toString());
-      }
-    } else {
+    // `compareOnlyIon` requires that the provided value be an ion.dom.Decimal instance.
+    if (other instanceof Decimal) {
+      isSupportedType = true;
+      valueToCompare = other.decimalValue();
+    }
+    if (options.coerceNumericType === true && other instanceof Float) {
+      isSupportedType = true;
+      valueToCompare = new ion.Decimal(other.toString());
+    }
+    if (!options.onlyCompareIon) {
```

Comment:
I think these can be `else if`s too. (I'll stop adding this comment from here on.)